patch-core.js

```diff
@@ -11,12 +11,35 @@ const https = require('https');
 const patchMarker = "__agent_base_https_request_patched__";
 if (!https.request[patchMarker]) {
   https.request = (function(request) {
-    return function(_options, cb) {
-      let options;
+    return function(_url, _options, cb) {
+      // Support for both
+      // https.request(options[, callback])
+      // https.request(url[, options][, callback])
+      if (typeof _url !== 'string' && !cb) {
+        cb = _options;
+        _options = _url;
+        _url = undefined;
+      }
+      if (!_url && _options) {
+        _url = url.format(_options);
+      }
+      var options;
       if (typeof _options === 'string') {
         options = url.parse(_options);
       } else {
-        options = Object.assign({}, _options);
+        options = url.parse(_url);
```

Comment:
This case feels unsafe as we simply discard the original value of `_options`. What is the assumption we are making here that gives us the ability to discard the old value?

Reply:
At this point we're sure to have a `_url`. What we're doing next is to grab the options out of the `_url` and merge them with the received options. In `https.get` we had this line `options = Object.assign({}, url.parse(_url), _options)` that had the same intentions of merging both _url and `options` into one object. What I'm doing is to make it work with the new WHATWG structure, as well as to make sure that both https.get and https.request follow this behavior.